### PR TITLE
[node] zlib: move constants into zlib.constants

### PR DESCRIPTION
**Pull request by RafaelKr**
Please fill in this template.

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).
- [x] Run `tsc` without errors.
- [x] Run `npm run lint package-name` if a `tslint.json` is present.

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:
[nodejs/node PR] [zlib: refactor constants for consistency](https://github.com/nodejs/node/pull/7203)
[Release Notes] https://nodejs.org/en/blog/release/v7.0.0/

### CODE REVIEW

File: node/index.d.ts [hunk 1]
```diff
@@ -1076,6 +1076,40 @@ declare module "zlib" {
...
+      Z_BINARY: number;
+      Z_TEXT: number;
+      Z_ASCII: number;
+      Z_UNKNOWN: number;
+      Z_DEFLATED: number;
+      Z_NULL: number;
+    };
+
     // Constants
     export var Z_NO_FLUSH: number;
```
Comment (sheetalkamat) [inline node/index.d.ts]: If they are exported as part of constants, do you need these definitions?

File: node/index.d.ts [hunk 2]
```diff
@@ -1076,6 +1076,40 @@ declare module "zlib" {
     export function unzip(buf: Buffer, callback: (error: Error, result: Buffer) => void): void;
     export function unzipSync(buf: Buffer, options?: ZlibOptions): Buffer;
 
+    export var constants: {
```
Comment (sheetalkamat) [inline node/index.d.ts]: May be you should make this namespace and the actual constants like Z_NO_FLUSH as export var declarations

**Review by sheetalkamat — CHANGES_REQUESTED** [review]
Leave a comment

File: node/index.d.ts (hunk 1 above)
Comment (RafaelKr) [inline node/index.d.ts]: I'm not sure. They still can be accessed.

Qutoe from the docs https://nodejs.org/api/zlib.html#zlib_constants:
`Note: Previously, the constants were available directly from require('zlib'), for instance zlib.Z_NO_FLUSH. Accessing the constants directly from the module is currently still possible but should be considered deprecated.`

File: node/index.d.ts (hunk 2 above)
Comment (RafaelKr) [inline node/index.d.ts]: In the description of the nodejs/node PR the author says
`refactor zlib constants for consistency with the changes to constants in fs, os`

the os module was declared directly below `zlib`, so I copied it from there. But I think you're right. I'll do it again and copy it from fs-module declaration.

**Comment by RafaelKr** [comment]
I put the constants into a namespace now as @sheetalkamat suggested.

I also noticed, that following constants were removed:
`Z_BINARY, Z_TEXT, Z_ASCII, Z_UNKNOWN, Z_DEFLATED, Z_NULL`
So I also didn't include them in the namespace. Should I also remove the following block?
```
export var Z_BINARY: number;
export var Z_TEXT: number;
export var Z_ASCII: number;
export var Z_UNKNOWN: number;
export var Z_DEFLATED: number;
export var Z_NULL: number;
```

Or would it be best, if I completely remove the constants which are declared by `export var ...` directly?

**Review by RafaelKr — COMMENTED** [review]
Review needed

File: node/index.d.ts (hunk 1 above)
Comment (sheetalkamat) [inline node/index.d.ts]: Its still unclear if these constant definitions need to be removed and replaced with the namespace constants or they both exist.

Comment (mhegazy) [inline node/index.d.ts]: they both exist.

File: node/index.d.ts [hunk 3]
```diff
@@ -1076,6 +1076,40 @@ declare module "zlib" {
...
+      Z_HUFFMAN_ONLY: number;
+      Z_RLE: number;
+      Z_FIXED: number;
+      Z_DEFAULT_STRATEGY: number;
+      Z_BINARY: number;
+      Z_TEXT: number;
+      Z_ASCII: number;
+      Z_UNKNOWN: number;
+      Z_DEFLATED: number;
+      Z_NULL: number;
```
Comment (mhegazy) [inline node/index.d.ts]: Z_NULL does not see to exist see https://github.com/nodejs/node/commit/197a465280#diff-c245d87dba893de0b77c7574f0081633L282

Comment (mhegazy) [inline node/index.d.ts]: here is what i see at runtime:

```ts
{
    Z_NO_FLUSH: 0,
     Z_PARTIAL_FLUSH: 1,
     Z_SYNC_FLUSH: 2,
     Z_FULL_FLUSH: 3,
     Z_FINISH: 4,
     Z_BLOCK: 5,
     Z_OK: 0,
     Z_STREAM_END: 1,
     Z_NEED_DICT: 2,
     Z_ERRNO: -1,
     Z_STREAM_ERROR: -2,
     Z_DATA_ERROR: -3,
     Z_MEM_ERROR: -4,
     Z_BUF_ERROR: -5,
     Z_VERSION_ERROR: -6,
     Z_NO_COMPRESSION: 0,
     Z_BEST_SPEED: 1,
     Z_BEST_COMPRESSION: 9,
     Z_DEFAULT_COMPRESSION: -1,
     Z_FILTERED: 1,
     Z_HUFFMAN_ONLY: 2,
     Z_RLE: 3,
     Z_FIXED: 4,
     Z_DEFAULT_STRATEGY: 0,
     ZLIB_VERNUM: 4736,
     DEFLATE: 1,
     INFLATE: 2,
     GZIP: 3,
     GUNZIP: 4,
     DEFLATERAW: 5,
     INFLATERAW: 6,
     UNZIP: 7,
     Z_MIN_WINDOWBITS: 8,
     Z_MAX_WINDOWBITS: 15,
     Z_DEFAULT_WINDOWBITS: 15,
     Z_MIN_CHUNK: 64,
     Z_MAX_CHUNK: Infinity,
     Z_DEFAULT_CHUNK: 16384,
     Z_MIN_MEMLEVEL: 1,
     Z_MAX_MEMLEVEL: 9,
     Z_DEFAULT_MEMLEVEL: 8,
     Z_MIN_LEVEL: -1,
     Z_MAX_LEVEL: 9,
     Z_DEFAULT_LEVEL: -1
}
```

**Review by mhegazy — CHANGES_REQUESTED** [review]

**Pull request merged**